lib/job.js

```diff
@@ -43,6 +44,38 @@ class Job extends BaseModel {
     }
 
     /**
+     * Lazy load the list of secrets
+     * @property secrets
+     * @return {Promise}    Resolves to a list of secrets
+     */
+    get secrets() {
+        delete this.secrets;
+
+        const secretNames = hoek.reach(this.permutations, '0.secrets', { default: [] });
```

Comment:
Ahh, that's because secrets cannot be mutated right now, so all permutations will have the same list of secrets.